app/models/season/phase_switcher.rb

```diff
@@ -1,44 +1,60 @@
+# frozen_string_literal: true
+
 class Season::PhaseSwitcher
 
+  PHASES = %i(
+    fake_proposals_phase
+    fake_application_phase
+    fake_coding_phase
+    back_to_reality
+    ).freeze
+
   def self.season
     Season.current
   end
 
+  def self.destined(phase)
+    raise ArgumentError.new("#{phase} is not a valid phase") unless phase.in? PHASES
+    self.public_send(phase)
+  end
+
+  private
```

Comment:
Then maybe I am wrong. I always thought that they were the same, just a matter of personal preference. I also thought this were a service object 🙈.
Well, I have homework to do. 
Lots of lesson learned by refactoring away a case statement ;-)